examples/multiple-regions/index.ts

```diff
@@ -16,13 +16,13 @@ function createServer(region: aws.Region): aws.ec2.Instance {
         ingress: [
             { protocol: "tcp", fromPort: 80, toPort: 80, cidrBlocks: ["0.0.0.0/0"] },
         ],
-    }, { provider: provider });
-    
+    }, <pulumi.CustomResourceOptions>{ provider: provider });
```

Comment:
why do we need these casts now?

Reply:
tagging @pgavlin .  It looks like we split ResourceOptions into ResourceOptions and CustomResourceOptions.  This resource still takes ResourceOptions, but 'provider' is in CustomResourceOptions.  So, without this cast, typescript thinks something fishy is going on.

I'm running into this because i'm pulling in a new @pulumi/pulumi version.  I'm not sure what our expectation is supposed to be here.

Reply:
How close are we to landing https://github.com/pulumi/pulumi/issues/1802? I think that would have let the code just work as written.

Reply:
Absent that, I think you need to write

```
{ providers: { aws: provider } }
```

Reply:
This is custom resource, not a component resource.  so the `{ providers: { aws: provider } }` version seems... wrong.